### PR TITLE
Optimizely Custom Click and View Attributes

### LINKED ISSUE #9790: Add Custom Attribute to Optimizely Click Tracking for Features
**Is your feature request related to a problem? Please describe.**
Track the component click event so it can be filtered in Optimizley.

**Describe the solution you'd like**
Set up Optimizley to use a custom attribute to segment/filter by users who have clicked the component
Pass the custom attribute with the click event sent with `optimizely.track(component_clicks)`

API for track:
`track(eventKey: string, overrideUserId?: string | EventTags, overrideAttributes?: UserAttributes, eventTags?: EventTags): void Track an event to the Optimizely results backend`

**Describe alternatives you've considered**

**Testing notes**
View a page running an experiment
View each variation on the page
Click for component on each variation
Check the results can be filtered by the clicked attribute
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### LINKED ISSUE #9835: Add Custom Attribute to Optimizely View Tracking for Features
**Is your feature request related to a problem? Please describe.**
Track the component view event so it can be filtered in Optimizley.

**Describe the solution you'd like**
Set up Optimizley to use a custom attribute to segment/filter by users who have viewed the component
Pass the custom attribute with the click event sent with `optimizely.track(component_views)`

API for track:
`track(eventKey: string, overrideUserId?: string | EventTags, overrideAttributes?: UserAttributes, eventTags?: EventTags): void Track an event to the Optimizely results backend`

**Describe alternatives you've considered**

**Testing notes**
View a page running an experiment
View each variation on the page
Check the results can be filtered by the view attribute
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by HarveyPeachey**
Resolves #9790
Resolves #9835

**Overall change:**
Adds a custom attribute for Optimizely tracking, that will allow segmentation of results.

**Code changes:**

- Adds code to override attributes associated with the original optimizely object.
- Uses the component name extracted from the eventTrackingData, to give the custom attribute a unique object key.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by MuhammadAHussain** [comment]
Will we need to add the override attributes for the Optimizely view tracking/scroll depth tracking too?

**Comment by HarveyPeachey** [comment]
> Will we need to add the override attributes for the Optimizely view tracking/scroll depth tracking too?

@MuhammadAHussain I don't think so, because of the way that sessions and segmentations work in Optimizely https://support.optimizely.com/hc/en-us/articles/4410289519373-How-Optimizely-counts-conversions#section_4. Hopefully testing the ticket should show this working.

(It took a long time to find this, I still don't know if I fully understand how it works)

**Review by amoore108 — APPROVED** [review]

File: src/app/hooks/useViewTracker/index.jsx [hunk 1]
```diff
@@ -72,7 +72,16 @@ const useViewTracker = (props = {}) => {
 
         if (shouldSendEvent) {
           if (optimizely) {
-            optimizely.track('component_views');
+            const overrideAttributes = {
```
Comment (MuhammadAHussain) [inline src/app/hooks/useViewTracker/index.jsx:75]: Don't see any tests covering the override attributes. Is it worth writing/modifying the expected to see if it's called with the correct overrides?

**Review by MuhammadAHussain — APPROVED** [review]
LGTM 👍

**Review by Louis-Matsika — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
Testing:

Scenario 1 --

Goto http://localhost:7080/mundo/articles/c45965g63xno
On Pageload shows experiment Variation_2
Get 1 pageview
Scroll to the promo
Get 1 componentview
Refresh the page 3 times
Get 3 pageviews
Viewed component 2 times
Got 2 component views
In Optimo Viewed Features = True && Service = Mundo

Expected = 4 Pageviews 3 ComponentViews
Result = 4 Pageviews 3 ComponentViews --> Works as expected

Scenario 2 --

Goto http://localhost:7080/mundo/articles/c45965g63xno
On Pageload it shows experiment Variation_2
See 1 pageview in console
Scroll the page to see the promo
Get 1 component view
Click on one of the item in Promo
Get a click event + 1 Pageview
Refresh the page 2 times
Get 2 Pageviews
Refresh the page again
Scroll to the promo
Get 1 Pageview and 1 ComponentView
Click on Promo
Get 1 Pageview + 1 Component View + 1 Click
Expected = 6 Pageviews + 3 Component Views + 2 Clicks
Result = 6 Pageviews + 1 Component View + 1 Clicks --> Not working as expected

@JonBeeb tested the above scenario and we believe that is working as expected, happy for this to be merged..

In Optimo Clicked Features = True && Service = Mundo

**Pull request merged**